src/main/java/uk/gov/justice/digital/hmpps/keyworker/services/KeyworkerService.java

```diff
@@ -225,6 +226,32 @@ public void allocate(OffenderKeyworker allocation) {
         return repository.findByStaffId(staffId);
     }
 
+    public List<KeyworkerAllocationDetailsDto> getAllocationsForKeyworkerWithOffenderDetails(Long staffId) {
+        final List<OffenderKeyworker> allocations = repository.findByStaffId(staffId);
+
+        final List<KeyworkerAllocationDetailsDto> detailsDtoList = allocations.stream().map(allocation -> {
+
+            URI uri = new UriTemplate("/locations/description/{agencyId}/inmates?keywords={offenderNo}").expand(staffId, allocation.getOffenderNo());
+            final OffenderSummaryDto offenderSummaryDto = restTemplate.getForObject(uri.toString(), OffenderSummaryDto.class);
+            return KeyworkerAllocationDetailsDto.builder()
```

Comment:
I created a ConversionHelper class specifically for model to DTO conversion (and vice versa). You could put this in that class if you wanted.